### DISCUSSION TITLE
Help Needed: Unable to do REST API Download java.net.URISyntaxException: Illegal character in scheme name at index 24:

### DISCUSSION THREAD

**Discussion by emarcelcom** [NightscoutFoundation/xDrip#3770] [Bug reports] [answered]
I'm receiving error when uploading data to nightscout. Any help would be much appreciated. 
Unable to do REST API 2024-11-11 21:23:10 (7) Download java.net.URIS yntaxException: illegal character in scheme name at index 24:
See pictures attached. 
![uploader_status](https://github.com/user-attachments/assets/1009e877-35b5-498c-9755-488377286e46)
![carelink_follow_status](https://github.com/user-attachments/assets/1f35a331-dea4-4c06-9849-c39b8251e9c5)

Tested with URLs: 
https://xxxxx.10be.de/ 
https://xxxxx.10be.de/api/v1

**Comment by Navid200** [comment]
If you enter the hostname in a browser, does it show Nightscout?

**Reply by emarcelcom** [reply]
Yes :) The server is up and running.

**Reply by emarcelcom** [reply]
Is there a way to test it using curl or something? maybe my API key is in wrong format? I just generated random bunch of string

**Reply by Navid200** [reply]
If you use a browser on the phone that is experiencing the problem and enter the hostname in the browser on that phone, does it show Nightscout?

**Reply by emarcelcom** [reply]
yes it works from samrtphone and from PC. Server is up and running and accessible.

**Reply by emarcelcom** [reply]
I'm also testing against a self-hosted server and I get the same error.

**Reply by Navid200** [reply, marked as answer]
Just look at the following page and make sure you are using the correct syntax:  
https://navid200.github.io/xDrip/docs/Nightscout-Settings.html

**Reply by emarcelcom** [reply]
I guess the format should be: 
https://password@hostname/api/v1/
What I typed was: password@https://hostname/api/v1/
Stupid error. 
Problem SOLVED! 
Thanks for your support.

**Reply by Navid200** [reply]
You're more than welcome
